include/seqan3/alphabet/all.hpp

```diff
@@ -39,9 +39,121 @@
  *
  * \defgroup alphabet Alphabet
  *
- * The alphabet module contains different biological alphabets and related functionality.
+ * ## Introduction
+ *
+ * Alphabets are a core component in SeqAn, they enable us to represent the smallest
+ * unit of biological sequence data, e.g. a nucleotide or an amino acid.
+ *
+ * In theory, these could just be represented as a `char` and this is how many people
+ * perceive them, but it makes sense to use a smaller, stricter and well-defined alphabet
+ * in almost all cases, because:
+ *
+ *   * Most of our sequences alphabets are actually smaller and can possibly be **represented by less bits**
+ * than a `char`, e.g. a `char` always uses 8bits of memory, a DNA character *could* be represented by 2bits
+ * if you limit it to the characters 'A', 'C', 'G', 'T';
+ *   * you may want to prevent the user from selecting **letters that aren't part of that alphabet** or converting
+ * those to a special "unknown"-character, e.g. 'J' is not part of any nucleotide definition so people shouldn't
+ * be able to select 'J' and in case you include 'N' in your nucleotide definition you might want 'J' to be
+ * converted to 'N';
+ *   * from a programmer's point of view we are also often interested in the **rank of a letter** inside its
+ * alphabet, i.e. we need to be able to convert 'A', 'C', 'G', 'T' to `0`, `1`, `2`, `3` respectively.
+ *
+ * In SeqAn there are alphabet types for typical sequence alphabets like DNA and amino acid, but also
+ * for qualities, RNA structures and alignment gaps. In addition there are templates for combining alphabet
+ * type into new alphabets and wrappers for existing data types like the canonical `char`.
+ *
+ * These are defined in sub-modules of the alphabet module, the collaboration diagram at the top of this page
+ * provides an overview. Many provide features beyond that of the generic alphabet concept which we define below.
+ * The inheritance diagram of seqan3::alphabet_concept also provides a good overview of the more refined
+ * concepts and also the classes implementing them.
+ *
+ * ## The alphabet concept
+ *
+ * The seqan3::alphabet_concept defines the requirements a type needs to meet to be considered an alphabet
+ * by SeqAn, or in other words: you can expect certain properties and functions to be defined on
+ * all data types we call an alphabet.
+ *
+ * ### alphabet size
+ *
+ * All alphabets in SeqAn have a fixed size. It
+ * can be queried via the seqan3::alphabet_size metafunction and often also the `value_size` static
```

Comment:
often? not always? 
Since you just mentioned that certain properties and functions can be expected, I would clearly distinct `alphabet_size()` to be a requirement and `value_size` a feature (at least if it is correct :laughing:, I don't have the concept in mind).

Reply:
> often? not always?

no.

> Since you just mentioned that certain properties and functions can be expected,...

Well it says "(see below for "members VS free/global functions")" and it is clearly documented in that section, or not?